src/database/DataTypes/TemperatureBreach.js

```diff
@@ -44,7 +44,7 @@ export class TemperatureBreach extends Realm.Object {
   }
 
   affectedBatches(database) {
-    return this.location.batchesAtTime(database, this.endTimestamp);
+    return this.location.batchesDuringTime(database, this.startTimestamp, this.endTimestamp);
```

Comment:
I think `location` is optional? Maybe some defensive code here?

Reply:
Added here. 
31b2b08 

Note I've merged a whole bunch of PRs into this one so just look at that commit 🤣